sm/tedge_agent/src/main.rs

```diff
@@ -9,10 +13,20 @@ async fn main() -> Result<(), anyhow::Error> {
     let user_manager = UserManager::new();
 
     let tedge_config_location = tedge_config::TEdgeConfigLocation::from_default_system_location();
+    let config_repository = tedge_config::TEdgeConfigRepository::new(tedge_config_location);
+    let tedge_config = config_repository.load()?;
 
     initialise_logging();
 
-    let agent = agent::SmAgent::new("tedge_agent", user_manager, tedge_config_location);
+    let default_plugin_type = tedge_config.query_string_optional(SoftwarePluginDefaultSetting)?;
+    let tedge_config_path = config_repository
+        .get_config_location()
+        .tedge_config_root_path()
+        .to_path_buf();
+    let sm_agent_config = SmAgentConfig::default()
+        .with_default_plugin_type(default_plugin_type)
+        .with_sm_home(tedge_config_path);
```

Comment:
This code should be moved into a `SmAgentConfig::new(tedge_config_location)` method.